src/main/java/org/broadinstitute/hellbender/tools/exome/ReadCountCollectionUtils.java

```diff
@@ -312,12 +408,235 @@ public static String getSampleNameFromReadCounts(final ReadCountCollection count
+        long topTruncatedCounts = 0;
+
+        for (int i = 0; i < targetCount; i++) {
+            final double[] rowCounts = counts.getRow(i);
+            for (int j = 0; j < columnCount; j++) {
+                final double count = rowCounts[j];
+                totalCounts++;
+                if (count < bottomPercentileThreshold) {
+                    counts.setEntry(i, j, bottomPercentileThreshold);
+                    bottomTruncatedCounts++;
+                } else if (count > topPercentileThreshold) {
+                    counts.setEntry(i, j, topPercentileThreshold);
+                    topTruncatedCounts++;
+                }
+            }
+        }
+        if (topTruncatedCounts == 0 && bottomTruncatedCounts == 0) {
+            logger.info(String.format("None of the %d counts where truncated as they all fall in the non-extreme range [%.2f, %.2f]", totalCounts, bottomPercentileThreshold, topPercentileThreshold));
+        } else {
+            final double truncatedPercentage = ((double)(topTruncatedCounts + bottomTruncatedCounts) / totalCounts) * 100;
+            logger.info(String.format("Some counts, %d out of %d (%.2f%%), where truncated as they fall out of the non-extreme range [%.2f, %.2f]",
+                    topTruncatedCounts + bottomTruncatedCounts, totalCounts, truncatedPercentage, bottomPercentileThreshold, topPercentileThreshold));
+        }
+    }
+
+    /**
+     * Creates a new read-count collection that is a copy of the input but dropping columns with extreme median coverage.
+     *
+     * @param readCounts the input read-counts.
+     * @param extremeColumnMedianCountPercentileThreshold bottom percentile to use as an exclusion threshold.
+     * @return never {@code null}. It might be a reference to the input read-counts if
+     * there are no columns to be dropped.
+     */
+    public static ReadCountCollection removeColumnsWithExtremeMedianCounts(final ReadCountCollection readCounts, final double extremeColumnMedianCountPercentileThreshold, final Logger logger) {
+        final List<String> columnNames = readCounts.columnNames();
+        final RealMatrix counts = readCounts.counts();
+        final Median medianCalculator = new Median();
+        final double[] columnMedians = IntStream.range(0, counts.getColumnDimension())
+                .mapToDouble(col -> medianCalculator.evaluate(counts.getColumn(col))).toArray();
+
+        // Calculate thresholds:
+        final double bottomExtremeThreshold = new Percentile(extremeColumnMedianCountPercentileThreshold).evaluate(columnMedians);
+        final double topExtremeThreshold = new Percentile(100 - extremeColumnMedianCountPercentileThreshold).evaluate(columnMedians);
+
+        // Determine kept and dropped column sets.
+        final Set<String> columnsToKeep = new HashSet<>(readCounts.columnNames().size());
+        final Map<String, Double> columnsToDrop = new LinkedHashMap<>((int) (extremeColumnMedianCountPercentileThreshold * 4.0 / 100.0));
```

Comment:
This is actually a bug, good catch! I think @samuelklee has fixed it in PR #618. I will move the fixed updated read filters from #618 to `ReadCountCollectionUtils`. It will also address Issue #646.